internal/armada/metrics/domain_test.go

```diff
@@ -0,0 +1,95 @@
+package metrics
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	armadaresource "github.com/armadaproject/armada/internal/common/resource"
+)
+
+func TestRecordJobRuntime_ShouldReportRuntimeDurations(t *testing.T) {
+	jobMetricsRecorder := NewJobMetricsRecorder()
+	jobMetricsRecorder.RecordJobRuntime("cpu", "", time.Second)
```

Comment:
Reading this test, I think you are testing three cases in one function. Could you refactor this to be a table driven test?

Reply:
can you see now if this what you meant?

Reply:
https://dave.cheney.net/2019/05/07/prefer-table-driven-tests

I think you are just missing a few items but this looks good to me!

Reply:
Cool. I tried to make something similar to the article, but I found it a bit tricky in this context.